geode-core/src/main/java/org/apache/geode/management/internal/operation/RebalanceOperationPerformer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal.operation;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.management.operation.RebalanceOperation;
+import org.apache.geode.management.runtime.RebalanceResult;
+
+@Experimental
+public class RebalanceOperationPerformer
+    implements OperationPerformer<RebalanceOperation, RebalanceResult> {
+  @Override
+  public RebalanceResult perform(RebalanceOperation operation) {
+    try {
+      Thread.sleep(100);
+    } catch (InterruptedException ignore) {
+    }
+    return new RebalanceResult("fake success");
```

Comment:
Is this product code or test code in src/main?

Reply:
In a future PR we plan on the first async operation we add being rebalance. The Rebalance classes you see are just place holders.
@onichols-pivotal maybe it would be better to get rid of the fake Rebalance classes in the product and instead prove in a test that the framework allows the addition of an operation and performer. I think we saw that at least one of the managers was missing public methods (i.e. it had to add the rebalance performer in the manager's constructor). Having a public method on th manager that lets you register the op/performer seems better (for future extensions) and you could move these rebalance classes in the pr to a test.

Reply:
Remember to reset the interrupt bit on the Thread if InterruptedException was caught (assuming it matters for this Thread).

Reply:
removed from this PR :)